javascript/integration-tests/AddOffer.spec.ts

```diff
@@ -0,0 +1,139 @@
+
+
+describe('AddOffer', (): void => {
+  let api: ApiPromise;
+  let borrower, lender: KeyringPair;
+  let loanTerms: PalletCreditcoinLoanTerms;
+  let borrowerRegAddr, lenderRegAddr: RegisteredAddress;
+  let askGuid, bidGuid: string;
+  let askOrderId: PalletCreditcoinAskOrderId;
+  let bidOrderId: PalletCreditcoinBidOrderId;
+
+  const blockchain = 'Ethereum';
+  const expirationBlock = 100;
+
+  beforeEach(async () => {
+    process.env.NODE_ENV = 'test';
+
+    const provider = new WsProvider('ws://127.0.0.1:9944');
+
+    api = await ApiPromise.create({ provider });
+
+    const keyring = new Keyring({ type: 'sr25519' });
+
+    lender = keyring.addFromUri('//Alice', { name: 'Alice' });
+    borrower = keyring.addFromUri('//Bob', { name: 'Bob' });
+    const lenderAddress = randomEthAddress();
+    const borrowerAddress = randomEthAddress();
+
+    loanTerms = api.createType<PalletCreditcoinLoanTerms>('PalletCreditcoinLoanTerms', {
+      amount: 1_000,
+      interestRate: 100,
+      maturity: 10
+    });
+
+    // register address for lender
+    let addr = await registerAddressAsync(api, lenderAddress, blockchain, lender);
+    expect(addr).toBeTruthy();
+
+    if (addr) {
+      lenderRegAddr = addr;
+      expect(lenderRegAddr.addressId).toBeTruthy();
+      askGuid = Guid.newGuid().toString();
+    } else {
+      throw new Error("Lender address wasn't registered successfully");
+    }
+
+    // register address for borrower
+    addr = await registerAddressAsync(api, borrowerAddress, blockchain, borrower);
+    expect(addr).toBeTruthy();
+
+    if (addr) {
+      borrowerRegAddr = addr;
+      expect(borrowerRegAddr.addressId).toBeTruthy();
+      bidGuid = Guid.newGuid().toString();
+    } else {
+      throw new Error("Borrower address wasn't registered successfully");
+    }
+
+    // AddAskOrder
+    let result = await addAskOrderAsync(api, lenderRegAddr.addressId, loanTerms, expirationBlock, askGuid, lender);
+    expect(result).toBeTruthy();
+
+    if (result) {
+      askOrderId = result;
+    } else {
+      throw new Error("askOrderId not found");
+    }
+    // todo: ^^^^ this block needs to be a helper test function
+
+
+    // AddBidOrder
+    result = await addBidOrderAsync(api, borrowerRegAddr.addressId, loanTerms, expirationBlock, bidGuid, borrower);
+    expect(result).toBeTruthy();
+
+    if (result) {
+      bidOrderId = result;
+    } else {
+      throw new Error("bidOrderId not found");
+    }
+
+console.log(`***** DEBUG: ${askOrderId}, ${bidOrderId}`);
+  }, 120000);
+
+  afterEach(async () => {
+    await api.disconnect();
+  });
+
+  it('fee is min 0.01 CTC', async (): Promise<void> => {
+    return new Promise((resolve, reject): void => {
+      const unsubscribe = api.tx.creditcoin
+        .addOffer(askOrderId, bidOrderId, expirationBlock)
+        .signAndSend(lender, { nonce: -1 }, async ({ dispatchError, events, status }) => {
+          expect(dispatchError).toBeFalsy();
```

Comment:
`Received: {"module": {"error": 29, "index": 5}}`

Is there a reference on how to decode these errors via the api? We need a helper function for that.